### PR TITLE
Bug fix for identifying amino acid loss; simplified identification code slightly; tests for identifying proteoforms

### PR DESCRIPTION
**Pull request by acesnik**

### CODE REVIEW

File: ProteoformSuiteInternal/Proteoform.cs [hunk 1]
```diff
@@ -94,7 +94,8 @@ public List<ExperimentalProteoform> identify_connected_experimentals(List<PtmSet
                         null); //Experimental without theoretical reference
                 string theoretical_base_sequence = theoretical_base != null ? theoretical_base.sequence : "";
 
-                PtmSet best_addition = generate_possible_added_ptmsets(r.peak.possiblePeakAssignments, deltaM, mass_tolerance, all_mods_with_mass, theoretical_base, theoretical_base_sequence, 1)
+                List<PtmSet> possible_additions = r.peak.possiblePeakAssignments.Where(p => Math.Sign(p.mass) == Math.Sign(deltaM)).ToList();
```
Comment (lschaffer2) [inline ProteoformSuiteInternal/Proteoform.cs]: this should say || p.mass == 0

Comment (acesnik) [inline ProteoformSuiteInternal/Proteoform.cs]: I see. If it's close to 0, unmodified should be considered. Far away from it, unmodified shouldn't be in the list. Good catch!

Should we consider slight negative values when it's a small positive value? The sign check would preclude considering those.

Comment (acesnik) [inline ProteoformSuiteInternal/Proteoform.cs]: We could use a loose mass tolerance (~1 Da) instead of sign.

Comment (lschaffer2) [inline ProteoformSuiteInternal/Proteoform.cs]: I think the only PTM list close to 0 worth considering is deamidation right? And this should only be negative?

**Pull request merged**